### DISCUSSION TITLE
Creation templates pre-filled using URL query params

### DISCUSSION THREAD

**Discussion by bertbesser** [firefly-iii/firefly-iii#4786] [Firefly III]
Hi,

I was wondering if there is a simple templating feature already available in firefly-iii. My searches only produced discussions on advanced suggestions-logic, that will not make it into ffiii.

Has the following simpler solution already been discussed? For creation urls like `https://.../transactions/create/transfer` allow optional query params that will pre-fill form-fields, e.g.

```
https://.../transactions/create/transfer?description=lunch&amount=6.20&category=lunch&source_account=my_source_account&destination_account=my_destination_account&...
```

would open the form in the following state:

![image](https://user-images.githubusercontent.com/4770807/119227082-90e72080-bb0c-11eb-9554-36c651782601.png)

That would allow to maintain a custom collection of 'templates' as plain browser bookmarks.

I guess that the implementation would not interfere with existing functionality of ffiii ( 'just sugar on top' ). Template bookmarking would not have to be integrated into ffiii as well.

What do you think?

Regards and thx for ffiii!

**Comment by JC5** [comment]
Firefly III used to have something like this. One day it broke and nobody complained for like a year. So nah, I'm not going to add this again.

**Reply by bertbesser** [reply]
Thanks for the _quick_ response.

> [...] So nah, I'm not going to add this again.

A bummer, but I can understand. ✅ 

Just to get you correctly: You're not willing to implement it yourself? Or -- stronger -- you wouldn't even want to maintain the code even if sb. else implemented it?

**Reply by JC5** [reply]
I don't want it at all.

**Reply by bertbesser** [reply]
I was afraid that would be the case :-)

Thanks for clarifying.

And thanks again for ffiii, I really enjoy using it.
